### PR TITLE
feat: Add yaml constructor to include files

### PR DESCRIPTION
**Pull request by igor-kramer**
Sometimes we have a lot of similar configs with identical parameters (usually some mappings for KafkaConnect).
Works with env variables not comfortable, looks nice to have one yaml with mappings to include it content to other configs.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/configuration/yaml.py [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+
+yaml.add_constructor('!include', construct_include, IncludeLoader)
+
 
 class YamlConfigurationMechanism(ConfigurationMechanism):
     """Ability to load configuration from yaml files"""
 
     def load_config(self, config_fp: IO) -> dict:
-        return yaml.safe_load(config_fp)
+        return yaml.load(config_fp, Loader=IncludeLoader)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/configuration/yaml.py:37]: Unfortunately, `yaml.load` is generally not considered safe and has had a number of CVEs in the past - I would not be comfortable using it

**Review by hsheth2 — COMMENTED** [review]
I'm not fully convinced that an `!include` directive is a good thing - if you've already got multiple config files, it might be more straightforward to do something like this https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/examples/library/programatic_pipeline.py and get full control over the config object's construction.

On the cons side, letting a `datahub ingest` command read (and dump - since we log a somewhat scrubbed ingestion config in a few places) arbitrary files on the filesystem definitely feels like surprising behavior and could easily cause security issues for unsuspecting users - that's actually how pyyaml.load ended up having so many CVEs to begin with.

As such, I'm leaning towards not accepting this PR, but happy to discuss further.

File: metadata-ingestion/src/datahub/configuration/yaml.py (hunk 1 above)
Comment (igor-kramer) [reply metadata-ingestion/src/datahub/configuration/yaml.py:37]: As i know `safe_load` is just `load(stream, Loader=SafeLoader)`. My class is inherited from SafeLoader.

**Comment by hsheth2** [comment]
@igor-kramer I'm going to close this for now - I'm not convinced that the extra complexity / unexpectedness of having a `!include` directive is justified. I'd recommend using `Pipeline.create` directly for now

Happy to revisit if we find more use cases / get more requests for this.

**Pull request closed without merging**